environment/deployments/roundtable/main.tf

```diff
@@ -267,6 +267,82 @@ resource "google_service_account" "reaper_sa" {
   project      = module.project_factory.project_id
 }
 
+# WorkloadIdentityPool for workloads to use Workload Identity Federation
+# instead of maintaining long-lived service account creds.
+resource "google_iam_workload_identity_pool" "pool" {
+  workload_identity_pool_id = "main"
+  display_name              = "Main"
+  description               = "Main identity pool for this project"
+  project                   = module.project_factory.project_id
+}
+
+# A Workload Identity Provider for workloads running from GitHub actions.
+#
+# https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions
+#
+# Available GitHub OIDC claims to map:
+# https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token
+resource "google_iam_workload_identity_pool_provider" "github" {
+  workload_identity_pool_id          = google_iam_workload_identity_pool.pool.workload_identity_pool_id
+  workload_identity_pool_provider_id = "github"
+  display_name                       = "GitHub"
+  description                        = "GitHub Actions identity pool provider."
+  attribute_mapping = {
+    "google.subject"       = "assertion.sub"
+    "attribute.actor"      = "assertion.actor"
+    "attribute.aud"        = "assertion.aud"
+    "attribute.repository" = "assertion.repository"
+  }
```

Comment:
For this PR, I don't think we _need_ `actor` and `aud`, but they could be nice to have for debugging. `google.subject` is required, and we use `attribute.repository` in the principal that we use to allow the impersonation of the service account.